### ISSUE TITLE
More robust testing for AlphaDropout

### ISSUE THREAD

**Issue by ToucheSir** [FluxML/Flux.jl#1851]
We currently borrow from https://github.com/pytorch/pytorch/blob/v1.10.0/test/cpp/api/modules.cpp#L1337-L1338, but some sort of goodness-of-fit test would be more robust than a simple range. Sampling the layer outputs should have a negligible impact on test runtime.

**Comment by ToucheSir** [comment]
The best place to start would be the paper itself, https://arxiv.org/abs/1706.02515. The section "New Dropout Technique" talks about Alpha dropout and some of its unique properties. Testing for those would be a great way to ensure we're following the spec.

**Comment by aritropc** [comment]
Well, that's a teeny-tiny paper, lemme read the required section and I'll get back to the tests :)

**Comment by arcAman07** [comment]
I would love to also work on this issue, read the required parts. Cold I get some more reference for this

**Comment by darsnack** [comment]
I have highlighted the appropriate sections in the text below. We want to sample many `x` values and test whether the mean and variance after dropout matches the paper.
![image](https://user-images.githubusercontent.com/4811350/154804723-6d03ccfe-2736-403a-9a4b-e93a2400ded4.jpeg)

**Comment by arcAman07** [comment]
Cool will start working on it as well as soon as I get done with the contrastive loss function.

**Comment by CarloLucibello** [comment]
Done — `test/layers/normalisation.jl` now checks the distributional property from the paper (unit mean and variance preserved after alpha dropout, in both training and test mode, in addition to the PyTorch-derived range check). Closing.
